### ISSUE TITLE
Populate email array with users from a specific role

### ISSUE THREAD

**Issue by jamiemax** [gemstoneit/metafit2014#17]
in Blogger/BlogBundle/Admin/EmailAdmin.php - you have an array of email addresses called $user_email_array. at the moment it just has 3 test email addresses in it

The objective:
We want to populate this array with the email addresses of users who are assigned to a specific role. If you create a new Email entity in the admin CRUD you have the option to select a ROLE for that email. Essentially we want to send that email to the users who have that role.

Note: The role hierarchy is as follows:

ROLE_SUPER_ADMIN
ROLE_ADMIN
ROLE_COACH
ROLE_NEWCOACH
ROLE_USER

users can have many roles - when a new email is created we want to send it ONLY to users who have the selected role as their TOP role. So for example if, when creating an email, you select ROLE_NEWCOACH we would want to send it to a user who has roles: {ROLE_NEWCOACH, ROLE_USER} but not a user who has roles {ROLE_COACH, ROLE_NEWCOACH, ROLE_USER}

Basically, only the TOP level role counts.

**Comment by Hoannv** [comment]
I read source code. I suggest use https://packagist.org/packages/mlpz/mailchimp-bundle for mailchimp api instead of using wordpress plugins as current now.

I've already set up groups. I also suggest remove roles and add groups to email.

**Comment by Hoannv** [comment]
@jamiemax do you know why we must have two mailchimp libraries ?

        //new mailchimp API
        require_once(__DIR__ . '/../Drewm/MailChimp.php');
        $mail_chimp = new \MailChimp('69ac6457e09367c2b1f705c8c04dc08f-us9');

        //old mailchimp API
        require_once(__DIR__ . '/../mailchimp-api-class/MCAPI.class.php');
        require_once(__DIR__ . '/../mailchimp-api-class/config.inc.php'); //contains apikey
        $api = new \MCAPI($apikey);

**Comment by jamiemax** [comment]
one method that we needed was not working in the new API, so we had to also use the old one

I agree that we should use the mailchimp package, but as we need to get an email on on Sunday, please can you just write the code that adds all email addresses from a group to that $user_email_array?

**Comment by Hoannv** [comment]
yes, users are populated. I only remove some codes, function will be available soon.

**Comment by jamiemax** [comment]
we have two because there was an API call which didn't work with the new library, so i included the old one so we could use this method

I agree we should use the mail chimp package, but as we need to send an email on Sunday, can you just write the code that populates the $user_email_array with all the email addresses from the selected group

Cheers,
Jamie



On 11 Feb 2015, at 13:51, Hoan Nguyen <notifications@github.com<mailto:notifications@github.com>> wrote:


@jamiemax<https://github.com/jamiemax> do you know why we must have two mailchimp libraries ?

    //new mailchimp API
    require_once(__DIR__ . '/../Drewm/MailChimp.php');
    $mail_chimp = new \MailChimp('69ac6457e09367c2b1f705c8c04dc08f-us9');

    //old mailchimp API
    require_once(__DIR__ . '/../mailchimp-api-class/MCAPI.class.php');
    require_once(__DIR__ . '/../mailchimp-api-class/config.inc.php'); //contains apikey
    $api = new \MCAPI($apikey);

**Comment by Hoannv** [comment]
@jamiemax email function is implemented. You can start testing it.

**Comment by Hoannv** [comment]
@jamiemax: I realized that users are added to list each mail send. I think it's not correct. Users should be added to list on mailchimp when they register.

And I removed some mailchimp file and used mailchimp package.

**Comment by jamiemax** [comment]
thanks for this - we tried sending to a small group of users and it worked fine, however when we sent to a large group, i.e. around 3000, it came back with a 504 error. do you know why this would be?

**Comment by Hoannv** [comment]
follow #26
